src/averageAges.js

```diff
@@ -55,7 +65,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersList = people.filter(person => person.sex === 'f');
+  const childrenList = people.filter(person => person.mother
+    && mothersList.some(mother => mother.name === person.mother));
+  const sonsList = childrenList.filter(children => children.sex === 'm');
+
+  return onlyWithSon
+    ? sonsList.reduce((acc, child) => (
+      acc + child.born - mothersList.find(mother => (
+        mother.name === child.mother)).born
+    ), 0) / sonsList.length
+    : childrenList.reduce((acc, child) => (
+      acc + child.born - mothersList.find(mother => (
+        mother.name === child.mother)).born
+    ), 0) / childrenList.length;
```

Comment:
you can add a variable where you would define which list to use (childrenList or sonsList) and then add this variable to the `reduce` to make the code simpler